### PR TITLE
restructure content fetching from github

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: bin/get_all_the_diffs [hunk 1]
```diff
@@ -54,24 +54,49 @@ missing_in_plumbing = repos.reject{|repo| plumbing_modules.include?(repo)}
...
     next
   end
-  sync = YAML.load(response)
-  version = sync['modulesync_config_version']
-  { repo => version }
+  msyncs[repo] = YAML.load(response)
+  begin
+    response = open("https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.sync.yml")
+  rescue OpenURI::HTTPError
+    puts "something is fucked up with #{repo} and https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.sync.yml"
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: We should remove vulgarity from error messages. Some may appreciate it but it’s not fully inclusive.

File: bin/get_all_the_diffs [hunk 2]
```diff
@@ -54,24 +54,49 @@ missing_in_plumbing = repos.reject{|repo| plumbing_modules.include?(repo)}
...
+    modules_that_have_missing_secrets << repo
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: Is there a plan for this variable? I want to say it should be named more accurately (missing_sync?) but I don’t see it used anywhere at all.

File: bin/get_all_the_diffs [hunk 3]
```diff
@@ -90,5 +168,10 @@ puts '==========================================================================
 puts "We need to add the following modules to plumbing: #{missing_in_plumbing.join(', ')}" if missing_in_plumbing.count > 0
 puts "We need to do an initial modulesync on: #{really_need_an_initial_modulesync.join(', ')}" if really_need_an_initial_modulesync.count > 0
 puts "we need to do a normal modulesync on: #{need_another_sync.join(', ')}"
+puts "We need to check Puppets version_requirement on these modules because it is not equal to #{PUPPET_SUPPORT_RANGE}: #{modules_with_incorrect_puppet_version_range.join(', ')}" if modules_with_incorrect_puppet_version_range.count > 0
+puts "We need to add Puppets version_requirement to: #{modules_without_puppet_version_range.join(', ')}" if modules_without_puppet_version_range.count > 0
+puts "We need to adjust the list of operatingsystemrelease for Debian to #{DEBIAN_SUPPORT_RANGE} on: #{supports_wrong_debian_version.join(', ')}"
+puts "We need to adjust the list of operatingsystemrelease for Ubuntu to #{UBUNTU_SUPPORT_RANGE} on: #{supports_wrong_ubuntu_version.join(', ')}"
+puts "We need to adjust the list of operatingsystemrelease for CentOS to #{CENTOS_SUPPORT_RANGE} on: #{supports_wrong_centos_version.join(', ')}"
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: Should these have if guards as well?

File: bin/get_all_the_diffs [hunk 4]
```diff
@@ -54,24 +66,90 @@ missing_in_plumbing = repos.reject{|repo| plumbing_modules.include?(repo)}
...
+modules_that_have_missing_secrets = []
+msyncs = {}
+syncs = {}
+metadatas = {}
+modulesync_repos.each do |repo|
   begin
     response = open("https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.msync.yml")
   rescue OpenURI::HTTPError
-    puts "something is fucked up with #{repo} and https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.msync.yml"
+    puts "something is fucked up with #{repo} and https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.msync.yml" if DEBUG
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: Can we remove the vulgarity here, please? It’s going to be funny to some and off putting to others.

File: bin/get_all_the_diffs [hunk 5]
```diff
@@ -54,24 +66,90 @@ missing_in_plumbing = repos.reject{|repo| plumbing_modules.include?(repo)}
...
     modules_that_were_added_but_never_synced << repo
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: This does not seem to be used

File: bin/get_all_the_diffs [hunk 6]
```diff
@@ -54,24 +66,90 @@ missing_in_plumbing = repos.reject{|repo| plumbing_modules.include?(repo)}
...
   end
-  sync = YAML.load(response)
-  version = sync['modulesync_config_version']
-  { repo => version }
+  msyncs[repo] = YAML.load(response)
+  begin
+    response = open("https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.sync.yml")
+  rescue OpenURI::HTTPError
+    puts "something is fucked up with #{repo} and https://raw.githubusercontent.com/voxpupuli/#{repo}/master/.sync.yml" if DEBUG
+    modules_that_have_missing_secrets << repo
```
Comment (rnelson0) [inline bin/get_all_the_diffs]: This does not seem to be used

File: bin/get_all_the_diffs (hunk 1 above)
Comment (bastelfreak) [reply bin/get_all_the_diffs]: sorry. removed it

**Comment by rnelson0** [comment]
@bastelfreak what about the operating system release sections, will they spit out junk if there’s no issues?

**Comment by rnelson0** [comment]
It looks like you're done but still has WIP in the title, let me know if you're good to go or if I should revoke my review.

**Pull request merged**